### PR TITLE
Add a comma after the optional bracketed expression in qlog!()

### LINKED ISSUE #91: Add a comma after the optional bracketed expression in qlog!()
The current arrangement doesn't get formatted because it doesn't look like real code. Adding a comma will allow rustfmt to format this as code.

```
qinfo!([self] "blah");
```
becomes
```
qinfo!([self], "blah");
```

### PR DESCRIPTION
**Pull request by o0Ignition0o**
Fixes #91

### CODE REVIEW

File: neqo-common/src/log.rs [hunk 1]
```diff
@@ -46,26 +46,26 @@ macro_rules! qlog {
 }
 #[macro_export]
 macro_rules! qerror {
-    ([$ctx:expr] $($arg:tt)*) => ( ::neqo_common::qlog!(::log::Level::Error, $ctx, $($arg)*); );
+    ([$ctx:expr], $($arg:tt)*) => ( ::neqo_common::qlog!(::log::Level::Error, $ctx, $($arg)*); );
```
Comment (martinthomson) [inline neqo-common/src/log.rs]: There is a minimal change here that might have made this easier.

```suggestion
    ([$ctx:expr] $(,)? $($arg:tt)*) => ( ::neqo_common::qlog!(::log::Level::Error, $ctx, $($arg)*); );
```

File: neqo-http3/src/connection.rs [hunk 2]
```diff
@@ -579,15 +584,19 @@ impl Http3Connection {
         let mut found = false;
 
         if let Some(request_stream) = &mut self.request_streams_client.get_mut(&stream_id) {
-            qdebug!([label] "Request/response stream {} is readable.", stream_id);
+            qdebug!(
+                [label],
+                "Request/response stream {} is readable.",
```
Comment (martinthomson) [inline neqo-http3/src/connection.rs]: What a shame that the extra character causes rustfmt to reflow this...
```suggestion
                "Request/response stream {} is readable",
```

File: neqo-common/src/log.rs (hunk 1 above)
Comment (o0Ignition0o) [reply neqo-common/src/log.rs]: Oh does it make the comma optional ? 
TIL Thanks!

Comment (o0Ignition0o) [reply neqo-common/src/log.rs]: I'm going to propagate the changes everywhere then

**Comment by o0Ignition0o** [comment]
Hmm it seems like the compiler wont allow me to apply your suggestion : 
```
#[macro_export]
macro_rules! qerror {
    ([$ctx:expr]$(,)? $($arg:tt)*) => ( ::neqo_common::qlog!(::log::Level::Error, $ctx, $($arg)*); );
    ($($arg:tt)*) => ( { ::neqo_common::log::init(); ::log::log!(::log::Level::Error, $($arg)*); } );
}
```

```error: local ambiguity: multiple parsing options: built-in NTs tt ('arg') or 1 other option.
   --> neqo-transport/src/connection.rs:332:27
    |
332 |             qerror!([self], "set token in state {:?}", self.state);
    |                           ^

error: aborting due to previous error

error: Could not compile `neqo-transport`.
```
A less nice version seems to build though
```
#[macro_export]
macro_rules! qerror {
    ([$ctx:expr], $($arg:tt)*) => ( ::neqo_common::qlog!(::log::Level::Error, $ctx, $($arg)*); );
    ([$ctx:expr] $($arg:tt)*) => ( ::neqo_common::qlog!(::log::Level::Error, $ctx, $($arg)*); );
    ($($arg:tt)*) => ( { ::neqo_common::log::init(); ::log::log!(::log::Level::Error, $($arg)*); } );
}
```

@martinthomson Would you like me to apply this change ?

**Comment by martinthomson** [comment]
The `$(,)?` form works perfectly for me.  Using:

```
rustc 1.36.0 (a53f9df32 2019-07-03)
```

**Comment by o0Ignition0o** [comment]
Oh that's weird, I'll have a look at what the CI says, maybe my setup is wrong :/

**Comment by o0Ignition0o** [comment]
```
error: local ambiguity: multiple parsing options: built-in NTs tt ('arg') or 1 other option.
  --> neqo-crypto/src/agent.rs:81:33
   |
81 |     qinfo!([format!("{:p}", fd)], "got ALPN {:?}", alpn);
   |                                 ^

error: aborting due to previous error
```
CI seems to agree with me Oo, maybe I missunderstood something ?

**Comment by martinthomson** [comment]
I've kicked off a build for the docker image used in CI, which should pick up rustc 1.36.0.  Maybe that will correct this.

**Comment by o0Ignition0o** [comment]
```
ignition@ignition-linux ~/Projects/oss/mozilla/neqo $ rustc -V
rustc 1.36.0 (a53f9df32 2019-07-03)
```
I have the same version as you, yet It won't build for me either :/

**Comment by o0Ignition0o** [comment]
Duplicate ot #123 (which has tests so it s much better)

**Pull request closed without merging**